tests/e2e/actions.go

```diff
@@ -268,73 +391,222 @@ type SubmitConsumerAdditionProposalAction struct {
+	}
+
+	cfg, exists := tr.testConfig.chainConfigs[e2e.ChainID(chainID)]
+	if !exists {
+		log.Fatal("no chain config found for consumer chain", chainID)
+	}
+	if cfg.ConsumerId != "" && cfg.ConsumerId != e2e.ConsumerID(consumerId) {
+		log.Fatalf("chain '%s'registered already with a different consumer ID '%s'", chainID, consumerId)
+	}
+
+	// Set the new created consumer-id on the chain's config
+	cfg.ConsumerId = e2e.ConsumerID(consumerId)
+	tr.testConfig.chainConfigs[e2e.ChainID(chainID)] = cfg
+
+	return e2e.ConsumerID(consumerId)
+}
+
+// submitConsumerAdditionProposal initializes a consumer chain and submits a governance proposal
 func (tr Chain) submitConsumerAdditionProposal(
 	action SubmitConsumerAdditionProposalAction,
 	verbose bool,
 ) {
-	spawnTime := tr.testConfig.containerConfig.Now.Add(time.Duration(action.SpawnTime) * time.Millisecond)
 	params := ccvtypes.DefaultParams()
-	template := `
-	{
- "messages": [
-  {
-   "@type": "/interchain_security.ccv.provider.v1.MsgConsumerAddition",
-   "chain_id": "%s",
-   "initial_height": {
-    "revision_number": "%d",
-    "revision_height": "%d"
-   },
-   "genesis_hash": "%s",
-   "binary_hash": "%s",
-   "spawn_time": "%s",
-   "unbonding_period": "%s",
-   "ccv_timeout_period": "%s",
-   "transfer_timeout_period": "%s",
-   "consumer_redistribution_fraction": "%s",
-   "blocks_per_distribution_transmission": "%d",
-   "historical_entries": "%d",
-   "distribution_transmission_channel": "%s",
-   "top_N": %d,
-   "validators_power_cap": %d,
-   "validator_set_cap": %d,
-   "allowlist": %s,
-   "denylist": %s,
-   "authority": "cosmos10d07y265gmmuvt4z0w9aw880jnsr700j6zn9kn",
-   "allow_inactive_vals": %t,
-   "min_stake": "%d"
-  }
- ],
-"metadata": "ipfs://CID",
-"deposit": "%dstake",
-"title": "Propose the addition of a new chain",
-"summary": "Gonna be a great chain",
-"expedited": false
-}`
-	jsonStr := fmt.Sprintf(template,
-		string(tr.testConfig.chainConfigs[action.ConsumerChain].ChainId),
-		action.InitialHeight.RevisionNumber,
-		action.InitialHeight.RevisionHeight,
-		base64.StdEncoding.EncodeToString([]byte("gen_hash")),
-		base64.StdEncoding.EncodeToString([]byte("bin_hash")),
-		spawnTime.Local().Format(time.RFC3339Nano),
-		params.UnbondingPeriod,
-		params.CcvTimeoutPeriod,
-		params.TransferTimeoutPeriod,
-		params.ConsumerRedistributionFraction,
-		params.BlocksPerDistributionTransmission,
-		params.HistoricalEntries,
-		action.DistributionChannel,
-		action.TopN,
-		action.ValidatorsPowerCap,
-		action.ValidatorSetCap,
-		action.Allowlist,
-		action.Denylist,
-		action.AllowInactiveVals,
-		action.MinStake,
-		action.Deposit)
+	spawnTime := tr.testConfig.containerConfig.Now.Add(time.Duration(action.SpawnTime) * time.Millisecond)
 
-	//#nosec G204 -- bypass unsafe quoting warning (no production code)
-	proposalFile := "/consumer-addition.proposal"
+	Metadata := types.ConsumerMetadata{
```

Comment:
`Metadata` and `InitializationParameters` do not have to start with a capital letter (?)